### PR TITLE
Mention multi-condition whispers and clarify JWT behaviour

### PR DESCRIPTION
**Pull request by Mario54**
@mspensieri @alavers @dannytranlx

### CODE REVIEW

File: source/rest/index.md [hunk 1]
```diff
@@ -89,7 +90,7 @@ curl https://api.smooch.io/v1/appusers/c7f6e6d6c3a637261bd9656f \
 
 JSON Web Tokens (JWTs) are an industry standard authentication mechanism. The full specification is described [here](https://tools.ietf.org/html/rfc7519), and a set of supported JWT libraries for a variety of languages and platforms can be found at [http://jwt.io](http://jwt.io). To summarize, a JWT is composed of a header, a payload, and a signature. The payload contains information called *claims* which describe the subject to whom the token was issued.
 
-For added security when making calls on behalf of an app user, a `jwt` credential can optionally be specified instead of an `appToken`. However other APIs, such as `/v1/webhooks` always require a valid `jwt` credential.
+For added security when making calls on behalf of an app user, a `jwt` credential can optionally be specified instead of an `appToken`. After using a `jwt` to authenticate an app user for the `/appuser` or `/init` routes, it becomes no longer possible to authenticate the app user with an `appToken`.
```
Comment (mspensieri) [inline source/rest/index.md]: After using an `appUser` scoped jwt only

File: source/rest/index.md [hunk 2]
```diff
@@ -89,7 +90,7 @@ curl https://api.smooch.io/v1/appusers/c7f6e6d6c3a637261bd9656f \
 
 JSON Web Tokens (JWTs) are an industry standard authentication mechanism. The full specification is described [here](https://tools.ietf.org/html/rfc7519), and a set of supported JWT libraries for a variety of languages and platforms can be found at [http://jwt.io](http://jwt.io). To summarize, a JWT is composed of a header, a payload, and a signature. The payload contains information called *claims* which describe the subject to whom the token was issued.
 
-For added security when making calls on behalf of an app user, a `jwt` credential can optionally be specified instead of an `appToken`. However other APIs, such as `/v1/webhooks` always require a valid `jwt` credential.
+For added security when making calls on behalf of an app user, a `jwt` credential can optionally be specified instead of an `appToken`. After using a `jwt` with `appUser` scope to authenticate an app user for the `/appuser` or `/init` routes, it becomes no longer possible to authenticate that app user with an `appToken`.
```
Comment (alavers) [inline source/rest/index.md]: Oh man we really needed this addition badly :3

I think it's also worth re-iterating this specification (that once you JWT you can't go back) in the guide docs, under `Authenticating users (optional)`

**Comment by alavers** [comment]
Rest looks :+1:

File: source/rest/index.md (hunk 2 above)
Comment (dannytranlx) [inline source/rest/index.md]: I would even put that into a `<aside>` notice!

**Comment by dannytranlx** [comment]
:+1:

**Comment by mspensieri** [comment]
:+1:

File: source/rest/index.md (hunk 2 above)
Comment (Mario54) [inline source/rest/index.md]: @dannytranlx An aside in the REST docs or in the SDK docs?

Comment (Mario54) [inline source/rest/index.md]: @alavers Actually it looks like it's already in the SDK docs

> Securing a `userId` happens automatically by using a JWT for the first time. Once a JWT is used to authenticate an individual `userId` with Smooch, that specific `userId` will require a JWT credential in all future init or login calls made to Smooch.

Comment (dannytranlx) [inline source/rest/index.md]: @Mario54 In the REST docs, exactly where you just added this line!

Comment (alavers) [inline source/rest/index.md]: :+1:

Comment (Mario54) [inline source/rest/index.md]: Like that?

image

Comment (dannytranlx) [inline source/rest/index.md]: ye

**Pull request merged**